FirebaseStorage/Tests/Unit/StorageReferenceTests.swift

```diff
@@ -44,11 +44,43 @@ class StorageReferenceTests: XCTestCase {
     XCTAssertEqual(ref.root().description, "gs://bucket/")
   }
 
+  func testRootWithURLAPI() throws {
+    let url = try XCTUnwrap(URL(string: "gs://bucket/path/to/object"))
+    let ref = try storage!.reference(for: url)
+    XCTAssertEqual(ref.root().description, "gs://bucket/")
+  }
+
   func testRootWithNoPath() throws {
     let ref = storage!.reference(forURL: "gs://bucket/")
     XCTAssertEqual(ref.root().description, "gs://bucket/")
   }
 
+  func testMismatchedBucket() throws {
+    do {
+      let url = try XCTUnwrap(URL(string: "gs://bcket/"))
+      _ = try storage!.reference(for: url)
+    } catch let StorageError.bucketMismatch(string) {
+      XCTAssertEqual(string, "Provided bucket: `bcket` does not match the Storage " +
+        "bucket of the current instance: `bucket`")
+      return
+    }
+    XCTFail()
```

Comment:
This unit test may be good use for [`XCTAssertThrowsError(_:_:file:line:_:)
`](https://developer.apple.com/documentation/xctest/1500795-xctassertthrowserror).

Something like:
```swift
let url = try XCTUnwrap(URL(string: "gs://bcket/"))
XCTAssertThrowsError(try storage!.reference(for: url), "This was supposed to fail.") { error in
  XCTAssertEqual(string, "Provided bucket: `bcket` does not match the Storage " +
          "bucket of the current instance: `bucket`")
}
```